doc/nvgre_tunnel/nvgre_tunnel.md

```diff
@@ -0,0 +1,437 @@
+| DB           | Database                              |
+| API          | Application Programming Interface     |
+| SAI          | Switch Abstraction Interface          |
+| YANG         | Yet Another Next Generation           |
+| CLI          | Command-line interface                |
+
+### Overview 
+
+NVGRE is a network virtualization method that uses encapsulation and tunneling to create large numbers of virtual LANs (VLANs) for subnets that can extend across dispersed data centers and layer 2 (the data link layer) and  layer 3 (the network layer). The SONIC has no support of NVGRE tunnel feature and appropriate SAI implementation for it. 
+
+From the architecture point of view the new orchagent daemon will be added to cover the NVGRE functionality. New daemon should handle the configuration taken from config DB, and call an appropriate SAI API, which will create tunnel and tunnel maps.
+
+### Requirements
+
+This section describes the SONiC requirements for NVGRE feature.
+
+At a high level the following should be supported:
+
+- User should be able to create NVGRE tunnel
+- User should be able to create VLAN to VSID mapper entries for the NVGRE tunnel
+- Both VLAN and Bridge to VSID mappers should be supported by the NVGRE tunnel
+- Only the decapsulation mappers supported
+- YANG model should be created in order to auto-generate CLI by using the [SONiC CLI Auto-generation tool](https://github.com/Azure/SONiC/blob/master/doc/cli_auto_generation/cli_auto_generation.md).
+- CLI for NVGRE tunnel
+
+Counters for NVGRE Tunnel are out of scope of this design document.
+
+### Architecture Design 
+
+The new tables will be added to Config DB. Unless otherwise stated, the attributes are mandatory.
+
+#### NVGRE configDB table
+
+```
+NVGRE_TUNNEL|{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+
+NVGRE_TUNNEL_MAP|{{tunnel_name}}|{{tunnel_map}}
+    "vsid": {{vsid_id}}
+    "vlan": {{vlan_id}}
+```
+
+#### ConfigDB schemas
+
+```
+; Defines schema for NVGRE Tunnel configuration attributes
+key                                   = NVGRE_TUNNEL:name             ; NVGRE tunnel configuration
+; field                               = value
+SRC_IP                                = ipv4                          ; IPv4 source address
+
+;value annotations
+ipv4          = dec-octet "." dec-octet "." dec-octet "." dec-octet
+dec-octet     = DIGIT                     ; 0-9  
+                  / %x31-39 DIGIT         ; 10-99  
+                  / "1" 2DIGIT            ; 100-199  
+                  / "2" %x30-34 DIGIT     ; 200-249
+```
+
+```
+; Defines schema for NVGRE Tunnel map configuration attributes
+key                                   = NVGRE_TUNNEL:tunnel_name:name ; NVGRE tunnel configuration
+; field                               = value
+VSID                                  = DIGITS                        ; 1 to 16 million values
+VLAN                                  = 1\*4DIGIT                     ; 1 to 4094 Vlan id
+```
+
```

Comment:
Please provide sample config-DB configurations.

Reply:
Config DB sample already located in another HLD section.